ros_message/src/time.rs

```diff
@@ -150,14 +138,16 @@ impl Hash for Duration {
 
 impl fmt::Display for Duration {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        let data = self.nanos();
-        display_nanos(
-            &format!(
-                "{}{:010}",
-                if data.is_negative() { "-" } else { "" },
-                data.abs()
-            ),
+        write!(
             f,
+            "{}{}.{:09}",
+            if self.sec.is_negative() || self.nsec.is_negative() {
+                "-"
+            } else {
+                ""
+            },
+            self.sec.abs(),
+            self.nsec.abs()
```

Comment:
This is the detail I was referencing in the other comment. With sec and nsec being -1, -1 we expect -1.000000001, while with nsec being 1 instead we'd expect -0.999999999.

I guess it's a failure of the test suite that I didn't add this in, as that was the reason for my more complex `display_nanos` function.